### PR TITLE
Update DG: Design, Architecture, UI, Logic

### PR DESCRIPTION
**Pull request by dunliang0513**
## What

1. Update the UML diagrams
2. Update the link to our project repo
3. Add label for UML diagrams in the Design part

## Why

Keep the Developer Guide up-to-date

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -64,18 +72,31 @@ Each of the four main components (also shown in the diagram above),
...
+The `UI` component uses the JavaFx UI framework. 
+
+Steps to start making changes to the UI:
+1. The layout of these UI parts are defined in matching `.fxml` files that are in the [`src/main/resources/view`](https://github.com/AY2223S2-CS2103T-W14-4/tp/tree/master/src/main/resources/view) folder.
+2. The managing classes of these UI parts are defined in matching `.java` files in the [`src/main/java/seedu/address/ui`](https://github.com/AY2223S2-CS2103T-W14-4/tp/tree/master/src/main/java/seedu/address/ui) folder.
+3. The stylesheets for these UI parts can be found in `Extensions.css` and `Mainwindow.css`, except `UserProfilePanel` and `HelpWindow` with separate CSS files in matching `.css` files in the [`src/main/resources/view`](https://github.com/AY2223S2-CS2103T-W14-4/tp/tree/master/src/main/resources/view) folder.
```
Comment (FredericChow00) [inline docs/DeveloperGuide.md:97]: add in HMHero.css too

**Review by FredericChow00 — CHANGES_REQUESTED** [review]
left some comments!

File: docs/images/StorageClassDiagram.png
[the remark this answers is not in the document]
Comment (Merrickneo) [reply docs/images/StorageClassDiagram.png:1]: @dunliang0513 @FredericChow00 I've done it in my PR don't worry about the refactoring!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (dunliang0513) [reply docs/DeveloperGuide.md:97]: OK~

**Review by FredericChow00 — APPROVED** [review]
LGTM!

**Pull request merged**